src/tests/ftest/control/dmg_telemetry_io_latency.py

```diff
@@ -0,0 +1,415 @@
+
+        """
+        transfer_sizes = self.params.get("transfer_sizes", "/run/*")
+        self.iterations = self.params.get("repetitions", "/run/*")
+        self.container = []
+        verification_results = []
+        metrics_data = {}
+        ior_latency = {}
+        # disable verbosity
+        self.telemetry.dmg.verbose = False
+        test_metrics = TelemetryUtils.ENGINE_IO_LATENCY_FETCH_METRICS + \
+                       TelemetryUtils.ENGINE_IO_LATENCY_UPDATE_METRICS
+
+        for transfer_size in transfer_sizes:
+            ior_latency[transfer_size] = {}
+            self.add_pool(connect=False)
+            oclass = self.ior_cmd.dfs_oclass.value
+            self.add_containers(self.pool, oclass)
+            for operation in ["update", "fetch"]:
+                flags = self.params.get("F", "/run/ior/ior{}flags/".format(
+                    operation))
+                self.log.info(
+                    "<<< Start ior %s transfer_size=%s", operation, transfer_size)
+                self.ior_cmd.transfer_size.update(transfer_size)
+                self.ior_cmd.flags.update(flags)
+                self.ior_cmd.set_daos_params(
+                    self.server_group, self.pool, self.container[-1].uuid)
+                # Run ior command
+                ior_results = self.run_ior_with_pool(
+                        timeout=200, create_pool=False, create_cont=False)
+                ior_latency[transfer_size][operation] = self.get_ior_latency(ior_results)
+                if operation in "update":
+                    metrics_data.update(self.telemetry.get_io_metrics(
+                        TelemetryUtils.ENGINE_IO_LATENCY_UPDATE_METRICS))
+                else:
+                    metrics_data.update(self.telemetry.get_io_metrics(
+                        TelemetryUtils.ENGINE_IO_LATENCY_FETCH_METRICS))
+            # Destroy the container and the pool.
+            self.destroy_containers(containers=self.container[-1])
+            self.destroy_pools(pools=self.pool)
+        for transfer_size in transfer_sizes:
+            status_dict = self.verify_rpc_latency_metrics(
+                metrics_data, test_metrics, str(transfer_size))
+            for operation in ["update", "fetch"]:
+                if status_dict[operation]:
+                    verification_results.append(
+                        ["PASSED", "engine_io_latency_" + operation, transfer_size])
+                else:
+                    verification_results.append(
+                        ["FAILED", "engine_io_latency_" + operation, transfer_size])
+        errors = False
+        self.log.error("Summary of io latency min, max, mean and stddev comparison results:")
+        for item in verification_results:
+            self.log.info("  %s  %s  %s", item[0], item[1], item[2])
+            if item[0] == "FAILED":
+                errors = True
+        if errors:
+            self.fail("Test FAILED")
+
+    @skipForTicket("DAOS-8985")
```

Comment:
DAOS-8985 is resolved. Should this be removed?

Reply:
yes... i will update that